File: data-structures/arrays/python/array_1D_update.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def main():
    # Declare and initialise the list of words
    spelling_words = ["me", "do", "it", "her", "him"]

    # Store the original word at index 4
    old_word = spelling_words[4]

    # Update the word stored at index 4
    spelling_words[4] = "log"

    # Output the updated word
    updated_word = spelling_words[4]
    print(f"The word {old_word} has been changed to {updated_word}")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: data-structures/arrays/c-sharp/array_1D_update.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace AdaCodeSamples {
    class Arrays1D {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Declare and initialise the array of words
            string[] spellingWords = new string[5] { 
                "me", "do", "her", "it", "him"
            };

            // Update the last word stored at index 4
            spellingWords[4] = "log";

            // Output the updated word
            string updatedWord = spellingWords[4];
            Console.WriteLine($"The last word has been changed to {updatedWord}");
        }
    }
}